webapp/lib/charts.dart

```diff
@@ -249,7 +249,7 @@ class SystemEventsTimeseriesLineChartView {
     chart = chartjs.Chart(canvas.getContext('2d'), chartConfig);
   }
 
-  void updateChart(Map<String, Map<DateTime, num>> updatedCountsAtTimestampList, {String timeScaleUnit = 'day', num upperLimit}) {
+  void updateChart(Map<String, Map<DateTime, num>> updatedCountsAtTimestampList, {String timeScaleUnit = 'day', num upperLimit, Map<String, DateTime> periodLimits}) {
```

Comment:
I think I'd prefer passing in the limits via separate variables, same as we do for `upperLimit`. I suggest `xLowerLimit` and `xUpperLimit`.

We'll probably want to transform `upperLimit` into `yLowerLimit` and `yUpperLimit` in a follow up PR as well

Reply:
I've created the separate variables `xLowerLimit` and  `xUpperLimit`.

I'll change the previous `upperLimit` into `yLowerLimit` and `yUpperLimit` in a follow-up PR.